core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -292,26 +293,28 @@ protected Node visitShowRoles(ShowRoles node, Void context)
         {
             Optional<String> catalog = node.getCatalog().map(c -> c.getValue().toLowerCase(ENGLISH));
 
+            List<Expression> rows;
             if (node.isCurrent()) {
                 accessControl.checkCanShowCurrentRoles(session.toSecurityContext(), catalog);
                 Set<String> enabledRoles = catalog.map(c -> metadata.listEnabledRoles(session, c))
                         .orElseGet(() -> session.getIdentity().getEnabledRoles());
-                List<Expression> rows = enabledRoles.stream()
+                rows = enabledRoles.stream()
                         .map(role -> row(new StringLiteral(role)))
                         .collect(toList());
-                return simpleQuery(
-                        selectList(new AllColumns()),
-                        aliased(new Values(rows), "role_name", ImmutableList.of("Role")));
             }
             else {
                 accessControl.checkCanShowRoles(session.toSecurityContext(), catalog);
-                List<Expression> rows = metadata.listRoles(session, catalog).stream()
+                rows = metadata.listRoles(session, catalog).stream()
                         .map(role -> row(new StringLiteral(role)))
                         .collect(toList());
-                return simpleQuery(
-                        selectList(new AllColumns()),
-                        aliased(new Values(rows), "role_name", ImmutableList.of("Role")));
             }
+            List<String> columns = ImmutableList.of("Role");
+            if (rows.isEmpty()) {
+                return emptyQuery(columns);
+            }
+            return simpleQuery(
+                    selectList(new AllColumns()),
+                    aliased(new Values(rows), "role_name", columns));
```

Comment:
pre-exsiting, is the role_name alias meaningless?

Reply:
It depends. I think the value used here is wrong here. It is an alias for a relation. So possibly you could reference the relation via column names. So in this test is meaningless, but in general it is needed.